celements-model/src/main/java/com/celements/model/migration/DanglingPropertiesMigration.java

```diff
@@ -0,0 +1,175 @@
+
+  public static final String NAME = "DanglingPropertiesMigration";
+
+  @Requirement
+  private HibernateSessionFactory sessionFactory;
+
+  @Requirement
+  private IQueryExecutionServiceRole queryExecutor;
+
+  @Requirement
+  private IModelAccessFacade modelAccess;
+
+  @Requirement
+  private ModelUtils modelUtils;
+
+  @Requirement
+  private ModelContext context;
+
+  private InformationSchema informationSchema;
+
+  @Override
+  public String getName() {
+    return NAME;
+  }
+
+  @Override
+  public String getDescription() {
+    return "deletes dangling properties due to hash collisions from mapped tables";
+  }
+
+  /**
+   * getVersion is using days since 1.1.2010 until the day of committing this migration
+   * 15.04.2018 -> 3025 http://www.wolframalpha.com/input/?i=days+since+01.01.2010
+   */
+  @Override
+  public XWikiDBVersion getVersion() {
+    return new XWikiDBVersion(3026);
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public void migrate(SubSystemHibernateMigrationManager manager, XWikiContext context)
+      throws XWikiException {
+    LOGGER.info("deleting dangling properties for database [{}]", getDatabaseWithPrefix());
+    try {
+      for (Iterator<PersistentClass> iter = getHibConfig().getClassMappings(); iter.hasNext();) {
+        PersistentClass mapping = iter.next();
+        String table = mapping.getTable().getName();
+        String className = mapping.getEntityName();
+        if (validateTableAndLogData(table, className)) {
+          migrateTable(table, className);
+        }
+      }
+    } catch (Exception exc) {
+      LOGGER.error("Failed to delete dangling properties for database [{}]",
+          getDatabaseWithPrefix(), exc);
+      throw exc;
+    } finally {
+      clearMigrationData();
+    }
+  }
+
+  private boolean validateTableAndLogData(String table, String className) throws XWikiException {
+    if (hasInternalCustomMapping(table, className)) {
+      try {
+        TableSchemaData data = getInformationSchema().get(table);
+        String sql = getSelectSql(table, data.getPkColumnName(), className);
+        LOGGER.trace("[{}] select sql: {}", table, sql);
+        List<List<String>> result = queryExecutor.executeReadSql(sql);
+        if (result.size() > 0) {
+          logData(table, result);
+          return true;
+        } else {
+          LOGGER.info("[{}] skip table, no dangling properties to delete", table);
+        }
+      } catch (IllegalArgumentException iae) {
+        LOGGER.warn("[{}] skip table, no TableSchemaData", table, iae);
+      }
+    } else {
+      LOGGER.debug("[{}] skip table, class [{}] has no internal custom mapping", table, className);
+    }
+    return false;
+  }
+
+  private void logData(String table, List<List<String>> result) {
+    checkState(LOGGER.isInfoEnabled(), "logging on level 'INFO' disabled");
+    LOGGER.info("[{}] dangling properties:", table);
```

Comment:
please either log dangling data on warning level (because it gets deleted) or write the data to a special dangling-properties table.